Renaming the charming with docker file.

mbruzek opened the pull request:
Updating navigation to reflect this change.

This work resolves issue #11, and #13

chuckbutler commented on src/navigation.tpl:
```diff
@@ -54,6 +54,7 @@
                   </li>
                   <li class="section"><a class="header" href="howto.html">How to...</a>
                     <ul class="sub">
+                      <li><a href="howto-charm-with-docker.html">Charm with Docker</a></li>
```
This strikes me as a user centric how-to section. We should probably make a new How-To section targeted at developers w/ our developer how-to's.

mbruzek commented:
I added a developer howto section, which necessitated creating a new landing page, and updating the navigation.  I also moved the vagrant file because that was more developer centric.

chuckbutler commented:
:+1: LGTM
